Sources/VLCSettingsController.m

```diff
@@ -146,16 +153,20 @@ - (void)PAPasscodeViewControllerDidSetPasscode:(PAPasscodeViewController *)contr
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
 {
     IASKSpecifier *specifier  = [self.settingsReader specifierForIndexPath:indexPath];
-    UITableViewCell *cell = [super tableView:tableView cellForRowAtIndexPath:indexPath];
-    cell.backgroundColor = PresentationTheme.current.colors.settingsCellBackground;
-    cell.textLabel.textColor = PresentationTheme.current.colors.cellTextColor;
-    cell.detailTextLabel.textColor = PresentationTheme.current.colors.cellDetailTextColor;
-    if ([specifier.key isEqualToString:@"about"]) {
-        cell.accessibilityIdentifier = VLCAccessibilityIdentifier.about;
+    VLCSettingsTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:specifier.type];
+    if (!cell) {
+        cell = [VLCSettingsTableViewCell cellWithIdentifier:specifier.type target:self];
     }
+    [cell configureForSpecifier:specifier value:[self.settingsStore objectForKey:specifier.key]];
```

Comment:
value is too vague can we have a variable instead and a couple of checks that the object that you try to get out of there is really the type that you expect?

Reply:
All type checks are done by `configureForSpecifier` for different types of cells with functions from InAppSettingsKit :D